src/sentry/static/sentry/app/api.tsx

```diff
@@ -425,35 +425,33 @@ export class Client {
     // compatibility layer which mimics that of the jquery response objects.
     fetchRequest
       .then(async response => {
-        let responseJSON: any;
-        let responseText: any;
+        const {ok, status, statusText} = response;
 
-        // Try to get JSON out of the response no matter the status
-        try {
-          responseJSON = await response.json();
-        } catch {
-          // No json came out.. too bad
-        }
+        const emulatedJQueryXHR: any = {
+          status,
+          statusText,
+          responseJSON: undefined,
+          responseText: undefined,
+          getResponseHeader: (header: string) => response.headers.get(header),
+        };
 
         // Try to get text out of the response no matter the status
         try {
-          responseText = await response.text();
+          emulatedJQueryXHR.responseText = await response.text();
         } catch {
           // No text came out.. too bad
         }
 
-        const {ok, status, statusText} = response;
-
-        const emulatedJQueryXHR: any = {
-          status,
-          statusText,
-          responseJSON,
-          responseText,
-          getResponseHeader: (header: string) => response.headers.get(header),
-        };
+        // Try to get JSON out of the response no matter the status
+        try {
+          emulatedJQueryXHR.responseJSON = await response.json();
+        } catch {
+          errorHandler(emulatedJQueryXHR, statusText, 'Invalid JSON');
+          return;
```

Comment:
Would it be better to be able to reach lines 456 and 457, but change from `'Request not OK'` to `'Invalid JSON'` of the 3rd parameter of `errorHandler()`?